### ISSUE TITLE
Error while compiling serde_codegen on latest nightly

### ISSUE THREAD

**Issue by lfairy** [serde-rs/serde#627]
On rustc 1.15.0-nightly (7b3eeea22 2016-11-21):

```
   Compiling serde_codegen v0.8.18
error[E0425]: unresolved name `syntax::parse::token::intern`
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.8.18/src/lib.rs:110:9
    |
110 |         syntax::parse::token::intern("derive_Serialize"),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unresolved name

error[E0425]: unresolved name `syntax::parse::token::intern`
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.8.18/src/lib.rs:115:9
    |
115 |         syntax::parse::token::intern("derive_Deserialize"),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unresolved name

error: no field `node` on type `&syntax::ast::Attribute`
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.8.18/src/lib.rs:147:32
    |
147 |                     match attr.node.value.node {
    |                                ^^^^
...
180 |     shim!(Serialize ser::expand_derive_serialize);
    |     ---------------------------------------------- in this macro invocation

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 1 field
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.8.18/src/lib.rs:148:25
    |
148 |                         ast::MetaItemKind::List(ref name, _) if name == "serde" => {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 field, found 2
...
180 |     shim!(Serialize ser::expand_derive_serialize);
    |     ---------------------------------------------- in this macro invocation

error: no field `node` on type `&syntax::ast::Attribute`
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.8.18/src/lib.rs:147:32
    |
147 |                     match attr.node.value.node {
    |                                ^^^^
...
181 |     shim!(Deserialize de::expand_derive_deserialize);
    |     ------------------------------------------------- in this macro invocation

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 1 field
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.8.18/src/lib.rs:148:25
    |
148 |                         ast::MetaItemKind::List(ref name, _) if name == "serde" => {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 field, found 2
...
181 |     shim!(Deserialize de::expand_derive_deserialize);
    |     ------------------------------------------------- in this macro invocation

error: aborting due to 4 previous errors
```

**Comment by lfairy** [comment]
rust-lang/rust#37824 might be relevant here

**Comment by theduke** [comment]
Just ran into this.

What's the last nightly that this works with?

**Comment by brunoqc** [comment]
> What's the last nightly that this works with?

I think it's nightly-2016-11-06.

**Comment by theduke** [comment]
With 2016-11-06 I get other errors with _derive 0.8.18. (after cargo clean).

```
error: attribute must only have one argument
 --> /home/theduke/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_derive-0.8.18/src/lib.rs:9:3
  |
9 | #[proc_macro_derive(Serialize, attributes(serde))]
  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: attribute must only have one argument
  --> /home/theduke/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_derive-0.8.18/src/lib.rs:17:3
   |
17 | #[proc_macro_derive(Deserialize, attributes(serde))]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

**Comment by dtolnay** [comment]
Fixed in [serde_derive 0.8.19](https://crates.io/crates/serde_derive/0.8.19).

**Comment by lfairy** [comment]
That was quick, thank you!
